### PR TITLE
Fix hardcoded sha1sum command and compare sha1 value

### PR DESCRIPTION
**Pull request by ChinYikMing**

### CODE REVIEW

File: mk/external.mk [hunk 1]
```diff
@@ -23,18 +23,26 @@ QUAKE_DATA_VERIFY = echo "$(strip $$($(T)_DATA_SHA1))  $$@" | $(SHA1SUM) -c
...
 TIMIDITY_DATA_URL = http://www.libsdl.org/projects/mixer/timidity/timidity.tar.gz
 TIMIDITY_DATA = $(OUT)/timidity
-TIMIDITY_DATA_SHA1 = cdd30736508d26968222a6414f3beabc3b7a0725
+TIMIDITY_DATA_SHA1 = cf6217a5d824b717ec4a07e15e6c129a4657ca25
 TIMIDITY_DATA_EXTRACT = tar -xf $(notdir $($(T)_DATA_URL)) -C $(OUT)
-TIMIDITY_TMP_FILE = /tmp/timidity_sha1.txt
-TIMIDITY_DATA_VERIFY = echo "$(TIMIDITY_DATA_SHA1)" > $(TIMIDITY_TMP_FILE) | find $(TIMIDITY_DATA) -type f -print0 | sort -z | xargs -0 shasum | shasum | cut -f 1 -d ' '
+TIMIDITY_TMP_FILE1 = /tmp/timidity_sha1.txt
+TIMIDITY_TMP_FILE2 = /tmp/timidity_sha2.txt
+TIMIDITY_DATA_VERIFY = echo "$(TIMIDITY_DATA_SHA1)" > $(TIMIDITY_TMP_FILE1) \
```
Comment (jserv) [inline mk/external.mk]: Can you make it modular? That is, provide calls for zip and (tar.)gz respectively?

File: mk/external.mk [hunk 2]
```diff
@@ -1,41 +1,113 @@
+COMPRESSED_SUFFIX :=
+COMPRESSED_IS_ZIP :=
+COMPRESSED_IS_DIR :=
+EXTRACTOR :=
+VERIFIER :=
+
+# temporarily files to store correct SHA1 value and computed SHA1 value
+# respectively for verification of directory source
+SHA1_FILE1 := /tmp/sha1_file1
```
Comment (jserv) [inline mk/external.mk]: Use `mktemp` instead.

File: mk/external.mk [hunk 3]
```diff
@@ -1,41 +1,110 @@
...
+#$(1) : compressed source URL
+define download
+$(eval _ := $(shell curl --progress-bar -O -L -C - "$(strip $(1))"))
+endef
+
+#$(1) : compressed source(.zip or.gz)
+define extract
+$(eval COMPRESSED_SUFFIX := $(suffix $(1)))
+$(eval COMPRESSED_IS_ZIP := $(filter $(COMPRESSED_SUFFIX),.zip))
+$(eval _ := $(if $(COMPRESSED_IS_ZIP),                  \
```
Comment (jserv) [inline mk/external.mk]: Indent with spaces.

File: mk/external.mk [hunk 4]
```diff
@@ -1,41 +1,110 @@
...
+
+# $(1) : correct SHA1 value
+# $(2) : filename or directory path
+# note:
+# 1. for regular file, $(SHA1SUM) command's -c option generates OK for indicating a match
+# 2. for directory, cmp command outputs nothing for indicating a match
+define verify
+    $(eval COMPRESSED_IS_DIR := $(if $(wildcard $(2)/*),1,0))
+    $(eval _ := $(if $(filter 1,$(COMPRESSED_IS_DIR)),             \
+		(                                                          \
```
Comment (jserv) [inline mk/external.mk]: Can you reduce the number of spaces? That is, I would expect more compact indetions.

Comment (ChinYikMing) [reply mk/external.mk]: > Can you reduce the number of spaces? That is, I would expect more compact indetions.

Do you mean to not align the backslash?

Comment (jserv) [reply mk/external.mk]: > Do you mean to not align the backslash?

No, I meant something like this:
```
define extract
    $(eval COMPRESSED_SUFFIX := $(suffix $(1)))
    $(eval COMPRESSED_IS_ZIP := $(filter $(COMPRESSED_SUFFIX),.zip))
    $(eval _ := \
        $(if $(COMPRESSED_IS_ZIP), \
            ($(eval EXTRACTOR := unzip -d $(OUT) $(1))), \
            ($(eval EXTRACTOR := tar -xf $(1) -C $(OUT))) \
    ))
    $(eval _ := $(shell $(EXTRACTOR)))
endef
```

Comment (ChinYikMing) [reply mk/external.mk]: > > Do you mean to not align the backslash?
> 
> No, I meant something like this:
> 
> ```
> define extract
>     $(eval COMPRESSED_SUFFIX := $(suffix $(1)))
>     $(eval COMPRESSED_IS_ZIP := $(filter $(COMPRESSED_SUFFIX),.zip))
>     $(eval _ := \
>         $(if $(COMPRESSED_IS_ZIP), \
>             ($(eval EXTRACTOR := unzip -d $(OUT) $(1))), \
>             ($(eval EXTRACTOR := tar -xf $(1) -C $(OUT))) \
>     ))
>     $(eval _ := $(shell $(EXTRACTOR)))
> endef
> ```

Got it.

File: mk/external.mk [hunk 5]
```diff
@@ -1,41 +1,98 @@
...
+define download
+    $(eval _ := $(shell curl --progress-bar -O -L -C - "$(strip $(1))"))
+endef
+
+# $(1): compressed source(.zip or.gz)
+define extract
+    $(eval COMPRESSED_SUFFIX := $(suffix $(1)))
+    $(eval COMPRESSED_IS_ZIP := $(filter $(COMPRESSED_SUFFIX),.zip))
+    $(eval _ :=  \
+		$(if $(COMPRESSED_IS_ZIP), \
```
Comment (jserv) [inline mk/external.mk]: Use 4 spaces for indention.

**Comment by jserv** [comment]
Thank @ChinYikMing for contributing!

**Pull request merged**